### PR TITLE
[WIP] Route naming.

### PR DESCRIPTION
**Pull request by delgermurun**
I think it is just what we need for now.

### CODE REVIEW

File: client/router.js [hunk 1]
```diff
@@ -36,12 +37,20 @@ Router.prototype.route = function(path, options) {
   };
 
   this._routes.push(route);
+  if (options.name) {
+    this._pathsMap[options.name] = path;
```
Comment (arunoda) [inline client/router.js]: Can't we store the route by the name. Then we have the path object there.
What do you think?

File: client/router.js [hunk 2]
```diff
@@ -36,12 +37,20 @@ Router.prototype.route = function(path, options) {
...
+  }
+
   this._updateCallbacks();
 
   return route;
 };
 
 Router.prototype.path = function(pathDef, fields, queryParams) {
+  if (this._pathsMap[pathDef]) {
```
Comment (arunoda) [inline client/router.js]: Just thinking. Do we need to ever worry about a clash between route name and a path.

Comment (delgermurun) [inline client/router.js]: We don't care subscription name clash. And we don't care wrong subscription name on check readiness (just return false). So I think it's better to follow one convention.

If we care clash name and a path, we should also worry about clash between name with other names etc ...

What do you think?

File: client/router.js (hunk 1 above)
Comment (delgermurun) [inline client/router.js]: Yeah, we can but I can't find use case that accessing to route object to by name for now. So I prefer minimal approach.

When we really need it, it's easy to change.

Comment (arunoda) [inline client/router.js]: We need to when handling middlewares. When using `except` and `only` kind of hooks. So, I prefer to layout the foundation. No need to change name related code later on.

**Comment by delgermurun** [comment]
@arunoda updated.

**Comment by arunoda** [comment]
Awesome.
Let's add few more tests and we can merge this :)

**Comment by delgermurun** [comment]
okay.

**Comment by delgermurun** [comment]
@arunoda Added tests.

I am wondering whether we need `FlowRouter.getRoute/getRouteByName` API or not?

**Comment by arunoda** [comment]
I don't think, we need an API like that.

**Comment by delgermurun** [comment]
Oh merged already. I was thinking about squash commits into one commit :). Please squash it before merge into master.

@arunoda When will you merge it? Will you wait 2.X release or not?

**Comment by arunoda** [comment]
I'll squash it :)
I think we don't need to wait this for 2.x (this does not have any breaking
changes)

**Comment by delgermurun** [comment]
Great. Thanks.

**Comment by hpx7** [comment]
Is there a way to get the name of the current route?

**Comment by VladShcherbin** [comment]
@hpx7 `FlowRouter.current().route.name`

**Comment by hpx7** [comment]
But really we want a way to declare a fine-grained dependency on the route name (`FlowRouter.current()` is neither reactive nor fine grained). So we'd need something like `FlowRouter.getRouteName()` as part of the API.

**Comment by arunoda** [comment]
It's not reactive for sure. What do you mean by fine grained?

**Comment by hpx7** [comment]
What I'm really referring to is `FlowRouter.reactiveCurrent()`, which will get called when anything about the route changes. `FlowRouter.getRouteName()` would only depend on the route name.

**Comment by arunoda** [comment]
Hmm. Good idea.
Let me think a bit. Anyway, why do you need such an api?

**Comment by hpx7** [comment]
For something like this
```handlebars
<li class="{{#if atRoute 'home'}}active{{/if}}">
```
Right now `atRoute` uses `Router.current().route.getName()` (from `iron:router`)

**Comment by arunoda** [comment]
For now you can use our reactiveCurrent().route.name

Let's see what we can do about a separate api.

**Pull request merged**